### PR TITLE
Unify decimal points in number values

### LINKED ISSUE #678: Numbers
Let's keep 2 decimals for everything for now. There is a probably a better way, but for beta is should be good.
meaning that all numbers should have 2 decimals, even if those decimals are .00

### PR DESCRIPTION
**Pull request by xpaczka**
Resolves #678 

Made all numbers values have two decimal points (except for integers like population)

### CODE REVIEW

File: src/shared/utils/numbers.ts [hunk 1]
```diff
@@ -4,10 +4,14 @@ import { FixedPointNumber } from "shared/types/stake"
...
 export const separateThousandsByComma = (
-  value: number | bigint | string
+  value: number | bigint | string,
+  noDecimals: boolean = false
 ): string => {
   const adjustedValue = typeof value === "string" ? +value : value
-  return adjustedValue.toLocaleString("en-US", { maximumFractionDigits: 2 })
+
+  return adjustedValue.toLocaleString("en-US", {
+    minimumFractionDigits: noDecimals ? 0 : 2,
```
Comment (jagodarybacka) [inline src/shared/utils/numbers.ts:13]: Before we used `maximumFractionDigits` - don't we need to set both `minimum` and `maximum` fraction digits?

**Comment by jagodarybacka** [comment]
@VladUXUI could you take a look at the preview here? imo the decimal part here doesn't make the numbers more readable
image
image
image

File: src/shared/utils/numbers.ts (hunk 1 above)
Comment (xpaczka) [reply src/shared/utils/numbers.ts:13]: Not sure, but probably you are right - will make it in case

**Comment by VladUXUI** [comment]
Looks a bit wonky indeed. So difficult to figure this out since we don't know what "dust" is going to be yet.

Maybe only do decimals for transaction related actions? (stake/unstake) and leave the rest without decimals? Since decimals are likely less important in those.
Wdyt @xpaczka @jagodarybacka @Shadowfiend  ?

Don't have enough brainpower for this atm 🧠 so any ideas/inputs are well received

**Comment by Shadowfiend** [comment]
IMO:
- XP doesn't need decimals.
- TAHO/veTAHO should probably only have decimals when needed (there should be reference code in the extension if it hasn't been pulled over that will trim trailing zeros).
- We should limit precision to 2 or 4 decimals in the display with a `title` attribute that shows the full precision, and make max buttons do full balance if it's more precise.
- We should also do the `<...` approach if the balance is nonzero but small.

There's a world where if the display doesn't match full precision we should show `…` or so. Less high conviction about that.

**Comment by jagodarybacka** [comment]
Ok so my suggestions are:
- let's make sure changed done in this PR are not affecting fixes from https://github.com/tahowallet/dapp/pull/699
- I would say function that separates thousands shouldn't decide about the precision at all if that's possible, we should be passing a number already parsed with desired precision
- Let's not display `.00` endings at all unless that makes sense in a given place (and I think we don't have a place like this right now 🤔 )

**Comment by xpaczka** [comment]
Converting to draft, will come back later since it is not top prio

**Review by jagodarybacka — APPROVED** [review]
✅

**Pull request merged**